_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils_new.cpp

```diff
@@ -2312,6 +2312,19 @@ void MfxHwH264Encode::ConfigureTask(
         task.m_maxPBFrameSize = 0;
     }
 
+#ifdef MFX_ENABLE_H264_MAX_FRAME_SIZE
+    if (video.mfx.RateControlMethod == MFX_RATECONTROL_CQP)
+    {
+        task.m_repackMaxIFrameSize = extOpt3.RepackMaxFrameSizeI;
+        task.m_repackMaxPFrameSize = extOpt3.RepackMaxFrameSizeP;
+        task.m_repackMaxBFrameSize = extOpt3.RepackMaxFrameSizeB;
+        task.m_repackNumPasses = extOpt3.RepackNumPasses;
+        assert(task.m_repackNumPasses <= 4); // for AVC encoder, NumPasses should be less than or equal to 4
```

Comment:
this check won't be present in release build. I believe we need it. To make it correctly you need to check ```extOpt3.RepackNumPasses``` somewhere in ```QueryVideoParamsQueryLike```

Reply:
Moved this check to CheckVideoParamQueryLike function. Thanks!